### PR TITLE
pylint: add module

### PR DESCRIPTION
**Pull request by florpe**
### Description

Adds a module for the Python linter pylint.

### Checklist

- [x] Change is backwards compatible.

- [ ] Code formatted with `./format`.

- [ ] Code tested through `nix-shell --pure tests -A run.all`.

- [ ] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [x] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [x] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

**Comment by berbiche** [comment]
Hi,

This module allows you to configure a global pylintrc configuration file, but can you override this file on a per-project basis without unsetting `PYLINTRC`?

According to this StackOverflow [answer](https://stackoverflow.com/a/32672068), it's possible to set the file `~/.pylintrc` with the user configuration.
IMO this would be a better solution than setting an environment variable as the environment variable will only work correctly if the user has set `programs.$SHELL.enable` for each shell.

**Comment by florpe** [comment]
According to the [pylint docs](https://pylint.pycqa.org/en/latest/user_guide/run.html#command-line-options), setting `PYLINTRC` or linking from `~/.pylintrc` will have the same effect - in either case `$PWD/.pylintrc` will take precedence. I see your point about shell integration though, and looking at other modules writing a dotfile seems like the more popular approach anyway, so I'll adjust the PR.

Is there a style guide for decisions like this?

**Comment by berbiche** [comment]
Could you rebase your changes on the latest "master" to fix the failing test case?

> and looking at other modules writing a dotfile seems like the more popular approach anyway, so I'll adjust the PR.
> 
> Is there a style guide for decisions like this?


AFAIK we don't currently have any documentation regarding when to use dotfiles or environment variables.
Dotfiles are generally a better idea because they work without needing to source Home Manager's environment variables.
Sometimes we use environment variables by wrapping tools with a script that sets the right environment before starting.

**Review by berbiche — CHANGES_REQUESTED** [review]

File: modules/programs/pylint.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+let
+
+  cfg = config.programs.pylint;
+  listToValue = concatMapStringsSep ",\n    " (generators.mkValueStringDefault { });
+  iniFormat = pkgs.formats.ini { inherit listToValue; };
+in {
+  meta.maintainers = [ hm.maintainers.florpe ];
+  options.programs.pylint = {
+    enable = mkEnableOption "The pylint Python linter.";
```
Comment (berbiche) [inline modules/programs/pylint.nix:13]:
```suggestion
    enable = mkEnableOption "the pylint Python linter";
```

`mkEnableOption` automatically appends a final period.

**Review by berbiche — CHANGES_REQUESTED** [review]
Small nit, lgtm besides that.

**Review by berbiche — CHANGES_REQUESTED** [review]
> ./modules/programs/pylint.nix: not formatted

Unfortunately the formatting script has to be run again.

**Comment by florpe** [comment]
It appears nixfmt has some difficulties with the newline in the list separator - now it's just `", "`, which is going to be ugly for longer lists, but at least the formatter is happy.

**Review by berbiche — APPROVED** [review]
LGTM!

**Pull request merged**